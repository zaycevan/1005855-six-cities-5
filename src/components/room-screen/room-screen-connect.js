import {connect} from "react-redux";
import {getNearOffers, getReviews, getAuthorizationStatus} from "@store/selectors";
import {fetchReviews, fetchNearOffers} from "@store/api-actions";
import {RoomScreen} from "./room-screen";

const mapStateToProps = (state) => ({
  nearOffers: getNearOffers(state),
  reviews: getReviews(state),
  authorizationStatus: getAuthorizationStatus(state),
});

const mapDispatchToProps = (dispatch) => ({
  loadReviews(offerId) {
    dispatch(fetchReviews(offerId));
  },
  loadNearOffers(offerId) {
    dispatch(fetchNearOffers(offerId));
  },
});

export default connect(mapStateToProps, mapDispatchToProps)(RoomScreen);
