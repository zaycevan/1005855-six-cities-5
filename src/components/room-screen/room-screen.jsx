import React, {PureComponent} from "react";
import {Link} from "react-router-dom";
import NearPlaceCard from "../near-place-card/near-place-card";
import ReviewFrom from "../review-form/review-form";
import {getPluralWord, formatReviewDate} from "../../utils/common";
import {offerPropType, offersPropType, reviewsPropType} from "../../utils/prop-types";

class RoomScreen extends PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      activeCard: ``,
    };
  }

  render() {
    const {offer, nearOffers, reviews} = this.props;
    const {id, photos, title, description, premium, type, rating, bedroomsCount, maxGuestsCount, price, amenities, hostInformation, reviewsId} = offer;
    const offerReviews = reviews.filter((review) => reviewsId.includes(review.id));

    if (photos.length > 6) {
      photos.length = 6;
    }

    return (
      <div className="page">
        <header className="header">
          <div className="container">
            <div className="header__wrapper">
              <div className="header__left">
                <Link to={`/`} className="header__logo-link">
                  <img className="header__logo" src="img/logo.svg" alt="6 cities logo" width="81" height="41"/>
                </Link>
              </div>
              <nav className="header__nav">
                <ul className="header__nav-list">
                  <li className="header__nav-item user">
                    <Link to={`/login`} className="header__nav-link header__nav-link--profile">
                      <div className="header__avatar-wrapper user__avatar-wrapper">
                      </div>
                      <span className="header__user-name user__name">Oliver.conner@gmail.com</span>
                    </Link>
                  </li>
                </ul>
              </nav>
            </div>
          </div>
        </header>

        <main className="page__main page__main--property">
          <section className="property" id={id}>
            <div className="property__gallery-container container">
              <div className="property__gallery">
                {photos.map((photo, i) => (
                  <div key={`${i}-${photo}`} className="property__image-wrapper">
                    <img className="property__image" src={photo} alt="Photo studio"/>
                  </div>
                ))}
              </div>
            </div>
            <div className="property__container container">
              <div className="property__wrapper">
                <div className="property__mark">
                  <span>{premium}</span>
                </div>
                <div className="property__name-wrapper">
                  <h1 className="property__name">
                    {title}
                  </h1>
                  <button className="property__bookmark-button button" type="button">
                    <svg className="property__bookmark-icon" width="31" height="33">
                      <use xlinkHref="#icon-bookmark"></use>
                    </svg>
                    <span className="visually-hidden">To bookmarks</span>
                  </button>
                </div>
                <div className="property__rating rating">
                  <div className="property__stars rating__stars">
                    <span style={{width: `${100 * rating / 5}%`}}></span>
                    <span className="visually-hidden">Rating</span>
                  </div>
                  <span className="property__rating-value rating__value">{rating}</span>
                </div>
                <ul className="property__features">
                  <li className="property__feature property__feature--entire">
                    {type}
                  </li>
                  <li className="property__feature property__feature--bedrooms">
                    {bedroomsCount} {getPluralWord(bedroomsCount, `Bedroom`, `Bedrooms`)}
                  </li>
                  <li className="property__feature property__feature--adults">
                    Max {maxGuestsCount} adults
                  </li>
                </ul>
                <div className="property__price">
                  <b className="property__price-value">&euro;{price}</b>
                  <span className="property__price-text">&nbsp;night</span>
                </div>
                <div className="property__inside">
                  <h2 className="property__inside-title">What&apos;s inside</h2>
                  <ul className="property__inside-list">
                    {amenities.map((amenity, i) => (
                      <li key={`${i}-${amenity}`} className="property__inside-item">
                        {amenity}
                      </li>
                    ))}
                  </ul>
                </div>
                <div className="property__host">
                  <h2 className="property__host-title">Meet the host</h2>
                  <div className="property__host-user user">
                    <div className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper">
                      <img className="property__avatar user__avatar" src={hostInformation.avatar} width="74" height="74" alt="Host avatar"/>
                    </div>
                    <span className="property__user-name">
                      {hostInformation.name}
                    </span>
                  </div>
                  <div className="property__description">
                    <p className="property__text">
                      {description}
                    </p>
                  </div>
                </div>
                <section className="property__reviews reviews">
                  <h2 className="reviews__title">Reviews &middot; <span className="reviews__amount">{offerReviews.length}</span></h2>
                  <ul className="reviews__list">
                    {offerReviews.map((offerReview) => (
                      <li key={offerReview.id} className="reviews__item">
                        <div className="reviews__user user">
                          <div className="reviews__avatar-wrapper user__avatar-wrapper">
                            <img className="reviews__avatar user__avatar" src={offerReview.avatar} width="54" height="54" alt="Reviews avatar"/>
                          </div>
                          <span className="reviews__user-name">
                            {offerReview.name}
                          </span>
                        </div>
                        <div className="reviews__info">
                          <div className="reviews__rating rating">
                            <div className="reviews__stars rating__stars">
                              <span style={{width: `${100 * offerReview.rating / 5}%`}}></span>
                              <span className="visually-hidden">Rating</span>
                            </div>
                          </div>
                          <p className="reviews__text">
                            {offerReview.text}
                          </p>
                          <time className="reviews__time" dateTime={offerReview.date}>{formatReviewDate(offerReview.date)}</time>
                        </div>
                      </li>
                    ))}
                  </ul>
                  <ReviewFrom />
                </section>
              </div>
            </div>
            <section className="property__map map"></section>
          </section>
          <div className="container">
            <section className="near-places places">
              <h2 className="near-places__title">Other places in the neighbourhood</h2>
              <div className="near-places__list places__list">
                {nearOffers.map((nearOffer) => (
                  <NearPlaceCard
                    key={nearOffer.id}
                    offer={nearOffer}
                    onCard={() => {
                      this.setState({activeCard: nearOffer});
                    }}
                  />
                ))}
              </div>
            </section>
          </div>
        </main>
      </div>
    );
  }
}

RoomScreen.propTypes = {
  offer: offerPropType.isRequired,
  nearOffers: offersPropType.isRequired,
  reviews: reviewsPropType.isRequired,
};

export default RoomScreen;
