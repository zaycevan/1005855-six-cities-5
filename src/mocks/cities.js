export default [`Paris`, `Cologne`, `Brussels`, `Amsterdam`, `Hamburg`, `Dusseldorf`];
