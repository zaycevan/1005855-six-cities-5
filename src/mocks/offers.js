import {HousingType} from "../const";
import {getRandomInteger} from "../utils/common";

const generateReviewsId = () => {
  const randomReviewsId = new Set();
  for (let i = 0; i < getRandomInteger(1, 6); i++) {
    randomReviewsId.add(getRandomInteger(11, 16));
  }

  return Array.from(randomReviewsId);
};

const generateRating = () => {
  const rating = (Math.floor((Math.random() * 5) * 10) / 10);
  return rating;
};

export default [
  {
    id: 1,
    photos: [`https://roomester.ru/wp-content/uploads/2018/08/skandinavskij-stil-v-interere-gostinoj-69.jpg`, `https://roomester.ru/wp-content/uploads/2019/03/gostinaya-spalnya-4-2.jpg`, `https://roomester.ru/wp-content/uploads/2019/02/gostinaya-v-stile-provans.jpg`, `https://roomester.ru/wp-content/uploads/2018/11/gostinaya-v-stile-loft.jpg`],
    title: `Lovely Guesthouse in the suburbs`,
    description: `Quiet and cosy tiny house in the suburbs of Amsterdam, just 10 minutes by metro from the city center of Amsterdam and 5 minutes away from Amsterdam Ajax Arena. The house is just 20 square meters, but it has everything you might need. It is located in a residential neighborhood, 2 minutes away from the metro station in a lovely green area. It is the ideal place for a couple.`,
    isPremium: Boolean(getRandomInteger(0, 1)),
    type: HousingType.APARTMENT,
    rating: generateRating(),
    bedroomsCount: getRandomInteger(1, 4),
    maxGuestsCount: getRandomInteger(2, 6),
    price: getRandomInteger(70, 300),
    amenities: [`wifi`, `dishwasher`, `parking`, `washer`, `elevator`, `conditioner`, `Baby seat`],
    coordinates: [52.3909553943508, 4.85309666406198],
    hostInformation: {
      avatar: `https://a0.muscache.com/im/pictures/user/9705ae17-cd68-4a1b-8668-46611c1811e7.jpg?im_w=240`,
      name: `Philip`,
      super: true
    },
    reviewsId: generateReviewsId()
  },
  {
    id: 2,
    photos: [`https://a0.muscache.com/im/pictures/fd1fa9b7-3459-4b4b-91ca-4d62772da7b3.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/af653a13-7892-4899-ae64-0481c5407cd5.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/6b95dbde-95a2-404a-814f-ac00845e2c7b.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/e01fd769-1fb4-41b1-8af7-0a435a4069b4.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/a0c674a1-0154-473d-abcc-bf2b432a392a.jpg?im_w=720`],
    title: `Central location private room all facillities`,
    description: `West town center. Private bright room 4x4 m, central heating, fully equipped kitchen, private shower+toilet. Third floor, stairs. WiFi, tv set, DVD, hairdryer. Towels, bed linen provided. Lively lovely neighborhood, quiet and safe at night. Street with characteristic Amsterdam houses and big trees.
    Museums, park, marke`,
    isPremium: Boolean(getRandomInteger(0, 1)),
    type: HousingType.ROOM,
    rating: generateRating(),
    bedroomsCount: getRandomInteger(1, 4),
    maxGuestsCount: getRandomInteger(2, 6),
    price: getRandomInteger(70, 300),
    amenities: [`wifi`, `dishwasher`, `Iron`, `TV`, `Washer`],
    coordinates: [52.369553943508, 4.85309666406198],
    hostInformation: {
      avatar: `https://a0.muscache.com/im/users/1232843/profile_pic/1317483378/original.jpg?im_w=720`,
      name: `Harm`,
      super: false
    },
    reviewsId: generateReviewsId()
  },
  {
    id: 3,
    photos: [`https://a0.muscache.com/im/pictures/3278447/0cec7b0a_original.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/3278451/c5c62f2c_original.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/3278456/130c852a_original.jpg?im_w=720`],
    title: `Private room in bed and breakfast hosted`,
    description: `Large split level room to acc. 3 people ( possible 4 but then a little crowded but cheaper!!) with fascilities as coffee/tea, radio, fridge, an ajacent bathroom and own toilet (extra toilet owner downstairs) ,microwave, wifi. Typical Amsterdamschool of architecture, good connection ( general transport) with inner city. Nice breakfast and warm hospitality`,
    isPremium: Boolean(getRandomInteger(0, 1)),
    type: HousingType.HOUSE,
    rating: generateRating(),
    bedroomsCount: getRandomInteger(1, 4),
    maxGuestsCount: getRandomInteger(2, 6),
    price: getRandomInteger(70, 300),
    amenities: [`wifi`, `Heating`, `Hair dryer`, `Laptop`],
    coordinates: [52.3909553943508, 4.929309666406198],
    hostInformation: {
      avatar: `https://a0.muscache.com/im/pictures/user/4af95a33-57b7-4f36-bc28-df74de058cc7.jpg?im_w=720`,
      name: `Erwin`,
      super: false
    },
    reviewsId: generateReviewsId()
  },
  {
    id: 4,
    photos: [`https://a0.muscache.com/im/pictures/12edb6ef-7cef-40ae-a01e-d98ec09d405a.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/b621e711-566e-4fd6-b11e-9a896c3b6aa6.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/1fe51232-1688-4cea-b56e-0ec83f26c0cf.jpg?im_w=720`, `https://a0.muscache.com/im/pictures/0623a3a7-3f6a-435e-a900-f5cc2ec750e0.jpg?im_w=720`],
    title: `Felix Super Apartment`,
    description: `Amazing Apartment in the middel of Amsterdam , only five minutes walking from Centraal Station and five minutes walking from Dam Square .`,
    isPremium: Boolean(getRandomInteger(0, 1)),
    type: HousingType.HOTEL,
    rating: generateRating(),
    bedroomsCount: getRandomInteger(1, 4),
    maxGuestsCount: getRandomInteger(2, 6),
    price: getRandomInteger(70, 300),
    amenities: [`wifi`, `Heating`, `TV`],
    coordinates: [52.3809553943508, 4.939309666406198],
    hostInformation: {
      avatar: `https://a0.muscache.com/im/pictures/user/2f3dfed9-fe3f-48d7-86dc-acf077d25711.jpg?im_w=720`,
      name: `Dario`,
      super: true
    },
    reviewsId: generateReviewsId()
  }
];
