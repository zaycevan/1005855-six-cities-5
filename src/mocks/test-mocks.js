import {NameSpace} from "@store/reducers/root-reducer";

export const TestMockData = {
  offer: {
    "city": `Cologne`,
    "cityLocation": [50.938361, 6.959974],
    "previewPhotos": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    "photos": [
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
      `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
    ],
    "title": `The house among olive `,
    "isFavorite": true,
    "isPremium": false,
    "rating": 3.8,
    "type": `house`,
    "bedroomsCount": 3,
    "maxGuestsCount": 4,
    "price": 493,
    "amenities": [
      `Laptop friendly workspace`,
      `Washer`,
      `Towels`
    ],
    "hostInformation": {
      "id": 25,
      "name": `Angelina`,
      "super": true,
      "avatar": `img/avatar-angelina.jpg`
    },
    "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    "coordinates": [50.957361, 6.9509739999999995],
    "id": 1
  },
  offers: [
    {
      "id": 1,
      "city": `Cologne`,
      "cityLocation": [50.938361, 6.959974],
      "previewPhotos": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      "photos": [
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`,
        `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`
      ],
      "title": `The house among olive `,
      "isFavorite": true,
      "isPremium": false,
      "rating": 3.8,
      "type": `house`,
      "bedroomsCount": 3,
      "maxGuestsCount": 4,
      "price": 493,
      "amenities": [
        `Laptop friendly workspace`,
        `Washer`,
        `Towels`
      ],
      "hostInformation": {
        "id": 25,
        "name": `Angelina`,
        "super": true,
        "avatar": `img/avatar-angelina.jpg`
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "coordinates": [50.957361, 6.9509739999999995]
    }
  ],
  offersFromServer: [
    {
      "bedrooms": 3,
      "city": {
        "location": {
          "latitude": 50.938361,
          "longitude": 6.959974,
          "zoom": 10
        },
        "name": `Cologne`
      },
      "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
      "goods": [`Laptop friendly workspace`, `Washer`, `Towels`],
      "host": {
        "avatar_url": `img/avatar-angelina.jpg`,
        "id": 25,
        "is_pro": true,
        "name": `Angelina`
      },
      "id": 1,
      "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
      "is_favorite": true,
      "is_premium": false,
      "location": {
        "latitude": 50.957361,
        "longitude": 6.9509739999999995,
        "zoom": 8
      },
      "max_adults": 4,
      "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
      "price": 493,
      "rating": 3.8,
      "title": `The house among olive `,
      "type": `house`
    },
  ],
  offerFromServer: {
    "bedrooms": 3,
    "city": {
      "location": {
        "latitude": 50.938361,
        "longitude": 6.959974,
        "zoom": 10
      },
      "name": `Cologne`
    },
    "description": `Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.`,
    "goods": [`Laptop friendly workspace`, `Washer`, `Towels`],
    "host": {
      "avatar_url": `img/avatar-angelina.jpg`,
      "id": 25,
      "is_pro": true,
      "name": `Angelina`
    },
    "id": 1,
    "images": [`https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg`, `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg`],
    "is_favorite": false,
    "is_premium": false,
    "location": {
      "latitude": 50.957361,
      "longitude": 6.9509739999999995,
      "zoom": 8
    },
    "max_adults": 4,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg`,
    "price": 493,
    "rating": 3.8,
    "title": `The house among olive `,
    "type": `house`
  },
  review: {
    "id": 1,
    "name": `Corey`,
    "avatar": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
    "authorId": 1,
    "authorSuper": true,
    "rating": 5,
    "text": `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    "date": new Date(`2020-11-01T13:38:44.752Z`)
  },
  reviews: [
    {
      "id": 1,
      "name": `Corey`,
      "avatar": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
      "authorId": 1,
      "authorSuper": true,
      "rating": 5,
      "text": `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
      "date": new Date(`2020-11-01T13:38:44.752Z`)
    }
  ],
  reviewsFromServer: [
    {
      "comment": `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
      "date": `2020-11-01T13:38:44.752Z`,
      "id": 1,
      "rating": 5,
      "user": {
        "avatar_url": `https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg`,
        "id": 1,
        "is_pro": true,
        "name": `Corey`
      }
    }
  ],
  user: {
    "email": `Oliver.conner@gmail.com`,
    "password": `12345678`
  },
  appStatus: `APP_READY`,
  currentCity: `Paris`,
  cities: [`Paris`, `Cologne`, `Brussels`],
  type: `Main`,
  currentSort: `POPULAR`
};

export const TestMockStore = {
  [NameSpace.CITIES]: {
    cities: TestMockData.cities,
    currentCity: TestMockData.currentCity,
  },
  [NameSpace.OFFERS]: {
    offers: TestMockData.offers,
    nearOffers: TestMockData.offers,
    favoriteOffers: TestMockData.offers,
    favoriteOfferStatus: `BEFORE_SENT`,
    favoriteOfferId: 1
  },
  [NameSpace.REVIEWS]: {
    reviews: TestMockData.reviews,
    reviewStatus: `BEFORE_SENT`
  },
  [NameSpace.SORT]: {
    currentSort: `POPULAR`,
  },
  [NameSpace.USER]: {
    user: TestMockData.user.email,
    authorizationStatus: `NO_AUTH`,
  },
  [NameSpace.APP]: {
    appStatus: `NOT_READY`,
  }
};
