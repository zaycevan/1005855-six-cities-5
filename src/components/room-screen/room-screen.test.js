import renderer from "react-test-renderer";
import {Provider} from "react-redux";
import {Router} from "react-router-dom";
import browserHistory from "@src/browser-history";
import configureStore from "redux-mock-store";
import {RoomScreen} from "./room-screen";
import {TestMockData, TestMockStore} from "@src/mocks/test-mocks";

const noop = () => {};
const mockStore = configureStore([]);
const store = mockStore(TestMockStore);

const tree = renderer.create(
    <Provider store={store}>
      <Router history={browserHistory}>
        <RoomScreen
          appStatus={`PAGE_READY`}
          nearOffers={TestMockData.offers}
          reviews={TestMockData.reviews}
          reviewStatus={`BEFORE_SENT`}
          authorizationStatus={`AUTH`}
          favoriteOfferStatus={`BEFORE_SENT`}
          loadData={noop}
          offer={TestMockData.offer}
          onFavoriteButtonClick={noop}
          activeCardId={1}
          onActiveCard={noop}
          outActiveCard={noop}
        />
      </Router>
    </Provider>
);

it(`Should RoomScreen render correctly`, () => {
  expect(tree.toJSON()).toMatchSnapshot();
});
