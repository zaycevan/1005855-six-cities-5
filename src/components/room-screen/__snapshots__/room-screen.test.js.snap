// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should RoomScreen render correctly 1`] = `
<div
  className="page"
>
  <header
    className="header"
  >
    <div
      className="container"
    >
      <div
        className="header__wrapper"
      >
        <div
          className="header__left"
        >
          <a
            className="header__logo-link "
            href="/"
            onClick={[Function]}
          >
            <img
              alt="6 cities logo"
              className="header__logo"
              height="41"
              src="img/logo.svg"
              width="81"
            />
          </a>
        </div>
        <nav
          className="header__nav"
        >
          <ul
            className="header__nav-list"
          >
            <li
              className="header__nav-item user"
            >
              <a
                className="header__nav-link header__nav-link--profile"
                href="/favorites"
                onClick={[Function]}
              >
                <div
                  className="header__avatar-wrapper user__avatar-wrapper"
                />
                <span
                  className="header__user-name user__name"
                >
                  Oliver.conner@gmail.com
                </span>
              </a>
            </li>
          </ul>
        </nav>
      </div>
    </div>
  </header>
  <main
    className="page__main page__main--property"
  >
    <section
      className="property"
      id={1}
    >
      <div
        className="property__gallery-container container"
      >
        <div
          className="property__gallery"
        >
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/15.jpg"
            />
          </div>
          <div
            className="property__image-wrapper"
          >
            <img
              alt="Photo studio"
              className="property__image"
              src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/16.jpg"
            />
          </div>
        </div>
      </div>
      <div
        className="property__container container"
      >
        <div
          className="property__wrapper"
        >
          <div
            className="property__name-wrapper"
          >
            <h1
              className="property__name"
            >
              The house among olive 
            </h1>
            <button
              className="property__bookmark-button button property__bookmark-button--active
                    "
              disabled={false}
              onClick={[Function]}
              type="button"
            >
              <svg
                className="property__bookmark-icon"
                height="33"
                width="31"
              >
                <use
                  xlinkHref="#icon-bookmark"
                />
              </svg>
              <span
                className="visually-hidden"
              >
                To bookmarks
              </span>
            </button>
          </div>
          <div
            className="property__rating rating"
          >
            <div
              className="property__stars rating__stars"
            >
              <span
                style={
                  Object {
                    "width": "76%",
                  }
                }
              />
              <span
                className="visually-hidden"
              >
                Rating
              </span>
            </div>
            <span
              className="property__rating-value rating__value"
            >
              3.8
            </span>
          </div>
          <ul
            className="property__features"
          >
            <li
              className="property__feature property__feature--entire"
            >
              house
            </li>
            <li
              className="property__feature property__feature--bedrooms"
            >
              3
               
              Bedrooms
            </li>
            <li
              className="property__feature property__feature--adults"
            >
              Max 
              4
               adults
            </li>
          </ul>
          <div
            className="property__price"
          >
            <b
              className="property__price-value"
            >
              €
              493
            </b>
            <span
              className="property__price-text"
            >
               night
            </span>
          </div>
          <div
            className="property__inside"
          >
            <h2
              className="property__inside-title"
            >
              What's inside
            </h2>
            <ul
              className="property__inside-list"
            >
              <li
                className="property__inside-item"
              >
                Laptop friendly workspace
              </li>
              <li
                className="property__inside-item"
              >
                Washer
              </li>
              <li
                className="property__inside-item"
              >
                Towels
              </li>
            </ul>
          </div>
          <div
            className="property__host"
          >
            <h2
              className="property__host-title"
            >
              Meet the host
            </h2>
            <div
              className="property__host-user user"
            >
              <div
                className="property__avatar-wrapper property__avatar-wrapper--pro user__avatar-wrapper"
              >
                <img
                  alt="Host avatar"
                  className="property__avatar user__avatar"
                  height="74"
                  src="img/avatar-angelina.jpg"
                  width="74"
                />
              </div>
              <span
                className="property__user-name"
              >
                Angelina
              </span>
            </div>
            <div
              className="property__description"
            >
              <p
                className="property__text"
              >
                Discover daily local life in city center, friendly neighborhood, clandestine casino, karaoke, old-style artisans, art gallery and artist studio downstairs.
              </p>
            </div>
          </div>
          <section
            className="property__reviews reviews"
          >
            <h2
              className="reviews__title"
            >
              Reviews · 
              <span
                className="reviews__amount"
              >
                1
              </span>
            </h2>
            <ul
              className="reviews__list"
            >
              <li
                className="reviews__item"
                id={1}
              >
                <div
                  className="reviews__user user"
                >
                  <div
                    className="reviews__avatar-wrapper user__avatar-wrapper"
                  >
                    <img
                      alt="Reviews avatar"
                      className="reviews__avatar user__avatar"
                      height="54"
                      src="https://assets.htmlacademy.ru/intensives/javascript-3/avatar/5.jpg"
                      width="54"
                    />
                  </div>
                  <span
                    className="reviews__user-name"
                  >
                    Corey
                  </span>
                </div>
                <div
                  className="reviews__info"
                >
                  <div
                    className="reviews__rating rating"
                  >
                    <div
                      className="reviews__stars rating__stars"
                    >
                      <span
                        style={
                          Object {
                            "width": "100%",
                          }
                        }
                      />
                      <span
                        className="visually-hidden"
                      >
                        Rating
                      </span>
                    </div>
                  </div>
                  <p
                    className="reviews__text"
                  >
                    The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.
                  </p>
                  <time
                    className="reviews__time"
                    dateTime={2020-11-01T13:38:44.752Z}
                  >
                    November 2020
                  </time>
                </div>
              </li>
            </ul>
            <form
              action="#"
              className="reviews__form form
    "
              method="post"
              onSubmit={[Function]}
            >
              <label
                className="reviews__label form__label"
                htmlFor="review"
              >
                Your review
              </label>
              <div
                className="reviews__rating-form form__rating"
              >
                <input
                  className="form__rating-input visually-hidden "
                  disabled={false}
                  id="5-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={5}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="5-stars"
                  title="perfect"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden "
                  disabled={false}
                  id="4-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={4}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="4-stars"
                  title="good"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden "
                  disabled={false}
                  id="3-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={3}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="3-stars"
                  title="not bad"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden "
                  disabled={false}
                  id="2-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={2}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="2-stars"
                  title="badly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
                <input
                  className="form__rating-input visually-hidden "
                  disabled={false}
                  id="1-stars"
                  name="rating"
                  onChange={[Function]}
                  type="radio"
                  value={1}
                />
                <label
                  className="reviews__rating-label form__rating-label"
                  htmlFor="1-stars"
                  title="terribly"
                >
                  <svg
                    className="form__star-image"
                    height="33"
                    width="37"
                  >
                    <use
                      xlinkHref="#icon-star"
                    />
                  </svg>
                </label>
              </div>
              <textarea
                className="reviews__textarea form__textarea"
                disabled={false}
                id="comment"
                maxLength={300}
                minLength={50}
                name="comment"
                onChange={[Function]}
                placeholder="Tell how was your stay, what you like and what can be improved"
                value=""
              />
              <div
                className="reviews__button-wrapper"
              >
                <p
                  className="reviews__help"
                >
                  To submit review please make sure to set 
                  <span
                    className="reviews__star"
                  >
                    rating
                  </span>
                   and describe your stay with at least 
                  <b
                    className="reviews__text-amount"
                  >
                    50 characters
                  </b>
                  .
                </p>
                <button
                  className="reviews__submit form__submit button"
                  disabled={true}
                  type="submit"
                >
                  Submit
                </button>
              </div>
            </form>
          </section>
        </div>
      </div>
      <section
        className="property__map map"
        id="map"
      />
    </section>
    <div
      className="container"
    >
      <section
        className="near-places places"
      >
        <h2
          className="near-places__title"
        >
          Other places in the neighbourhood
        </h2>
        <div
          className="near-places__list places__list"
        >
          <article
            className="near-places__card place-card"
            id={1}
            onMouseEnter={[Function]}
            onMouseLeave={[Function]}
          >
            <div
              className="near-places__image-wrapper place-card__image-wrapper"
            >
              <a
                href="/offer/1"
                onClick={[Function]}
              >
                <img
                  alt="Place image"
                  className="place-card__image"
                  height="200"
                  src="https://assets.htmlacademy.ru/intensives/javascript-3/hotel/14.jpg"
                  width="260"
                />
              </a>
            </div>
            <div
              className=" place-card__info"
            >
              <div
                className="place-card__price-wrapper"
              >
                <div
                  className="place-card__price"
                >
                  <b
                    className="place-card__price-value"
                  >
                    €
                    493
                  </b>
                  <span
                    className="place-card__price-text"
                  >
                    / night
                  </span>
                </div>
                <button
                  className="place-card__bookmark-button button place-card__bookmark-button--active
            "
                  disabled={false}
                  onClick={[Function]}
                  type="button"
                >
                  <svg
                    className="place-card__bookmark-icon"
                    height="19"
                    width="18"
                  >
                    <use
                      xlinkHref="#icon-bookmark"
                    />
                  </svg>
                  <span
                    className="visually-hidden"
                  >
                    In bookmarks
                  </span>
                </button>
              </div>
              <div
                className="place-card__rating rating"
              >
                <div
                  className="place-card__stars rating__stars"
                >
                  <span
                    style={
                      Object {
                        "width": "76%",
                      }
                    }
                  />
                  <span
                    className="visually-hidden"
                  >
                    Rating
                  </span>
                </div>
              </div>
              <h2
                className="place-card__name"
              >
                <a
                  href="/offer/1"
                  onClick={[Function]}
                >
                  The house among olive 
                </a>
              </h2>
              <p
                className="place-card__type"
              >
                house
              </p>
            </div>
          </article>
        </div>
      </section>
    </div>
  </main>
</div>
`;
